File: backend/src/main/java/com/devmatch/backend/domain/analysis/entity/AnalysisResult.java
package com.devmatch.backend.domain.analysis.entity;

import com.devmatch.backend.domain.application.entity.Application;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "analysis_results")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AnalysisResult {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @OneToOne(mappedBy = "analysisResult")
  private Application application;

  @Column(name = "compatibility_score", precision = 5, scale = 2, nullable = false)
  private BigDecimal compatibilityScore;

  @Column(columnDefinition = "TEXT", nullable = false)
  private String compatibilityReason;
}

File: backend/src/main/java/com/devmatch/backend/domain/analysis/entity/AnalysisResult.kt
package com.devmatch.backend.domain.analysis.entity

import com.devmatch.backend.domain.application.entity.Application
import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "analysis_results")
class AnalysisResult(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,

    @OneToOne(mappedBy = "analysisResult", fetch = FetchType.LAZY)
    var application: Application? = null,

    @Column(name = "compatibility_score", precision = 5, scale = 2, nullable = false)
    var compatibilityScore: BigDecimal = BigDecimal.ZERO,

    @Column(columnDefinition = "TEXT", nullable = false)
    var compatibilityReason: String = ""
) {
    // JPA를 위한 기본 생성자 (리플렉션으로 사용됨)
    private constructor() : this(
        0L,
        null,
        BigDecimal.ZERO,
        ""
    )
    
    companion object {
        fun create(
            application: Application,
            score: BigDecimal,
            reason: String
        ) = AnalysisResult(
            application = application,
            compatibilityScore = score,
            compatibilityReason = reason
        )
    }
}